### DISCUSSION TITLE
How do I add a memory address of a pointer as a span field?

### DISCUSSION THREAD

**Discussion by vi** [tokio-rs/tracing#2778] [Q&A] [answered]
How do I do something like `{:p}` format as a field of a span?

`tracing::field` contains helpers for formatting as `Debug` and as `Display`, but I don't see the [`std::fmt::Pointer`](https://doc.rust-lang.org/stable/std/fmt/trait.Pointer.html#) helper?

Ideally it should be stored as a number (without bulky allocations) and formatted as hex, just like `{:p}`.

**Comment by hawkw** [comment, marked as answer]
You can use `format_args!("{:p}", ...)` as a normal `Debug` or `Display` field. The `format_args!` macro produces a type which borrows the arguments and implements `Debug` and `Display`, rather than allocating and producing a `String`, like `format!`.

It might be worthwhile to add helper functions to `tracing::field` for producing `dyn Display`s that format something using `fmt::Poitner`, `fmt::UpperHex`/`fmt::LowerHex`, and other formatting traits. But, in the meantime, using `format_args!` should work.
